File: easy.server.demo/easy.server.tomcat.demo/src/main/java/io/github/jinlongliao/easy/server/demo/core/ServletDemo.java
package io.github.jinlongliao.easy.server.demo.core;

import io.github.jinlongliao.easy.server.demo.logic.param.UserModel;
import io.github.jinlongliao.easy.server.servlet.BaseHttpServlet;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;
import org.springframework.web.multipart.support.StandardMultipartHttpServletRequest;

import jakarta.servlet.MultipartConfigElement;
import jakarta.servlet.ServletContext;
import jakarta.servlet.ServletException;
import jakarta.servlet.ServletRegistration;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * @author: liaojinlong
 * @date: 2022/5/28 19:46
 */
@Component
public class ServletDemo extends BaseHttpServlet<UserModel> {
    @Override
    public String[] supportPath() {
        return new String[]{"/test"};
    }

    @Override
    protected void todoLogic(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        UserModel requestBody = getRequestBody(req);
        resp.getWriter().write(requestBody.toString());
    }

    @Override
    protected HttpServletRequest handleRequest(HttpServletRequest request) {
        if (StringUtils.startsWithIgnoreCase(request.getContentType(), ("multipart/form-data"))) {
            return new StandardMultipartHttpServletRequest(request);
        }
        return request;
    }

    @Override
    public void extraConfig(ServletRegistration.Dynamic dynamic, ServletContext servletContext) {
        super.extraConfig(dynamic, servletContext);
        dynamic.setMultipartConfig(new MultipartConfigElement(""));
    }

    @Override
    public String getDescription() {
        return "测试Servlet";
    }
}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/core/ServletDemo.kt
package io.github.jinlongliao.easy.server.kotlin.demo.core

import io.github.jinlongliao.easy.server.kotlin.demo.logic.param.UserModel
import io.github.jinlongliao.easy.server.servlet.BaseHttpServlet
import jakarta.servlet.MultipartConfigElement
import jakarta.servlet.ServletContext
import jakarta.servlet.ServletException
import jakarta.servlet.ServletRegistration
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import org.springframework.util.StringUtils
import org.springframework.web.multipart.support.StandardMultipartHttpServletRequest
import java.io.IOException

/**
 * @author: liaojinlong
 * @date: 2022/5/28 19:46
 */
@Component
class ServletDemo : BaseHttpServlet<UserModel?>() {
    override fun supportPath(): Array<String> {
        return arrayOf("/test")
    }

    @Throws(ServletException::class, IOException::class)
    override fun todoLogic(req: HttpServletRequest, resp: HttpServletResponse) {
        val requestBody = getRequestBody(req)
        resp.writer.write(requestBody.toString())
    }

    override fun handleRequest(request: HttpServletRequest): HttpServletRequest {
        return if (StringUtils.startsWithIgnoreCase(request.contentType, "multipart/form-data")) {
            StandardMultipartHttpServletRequest(request)
        } else request
    }

    override fun extraConfig(dynamic: ServletRegistration.Dynamic, servletContext: ServletContext) {
        super.extraConfig(dynamic, servletContext)
        dynamic.setMultipartConfig(MultipartConfigElement(""))
    }

    override fun getDescription(): String {
        return "测试Servlet"
    }
}
